huaweicloud/services/cdn/resource_huaweicloud_cdn_domain.go

```diff
@@ -1619,11 +1619,69 @@ func resourceCdnDomainDelete(ctx context.Context, d *schema.ResourceData, meta i
 		return diag.Errorf("error deleting CDN domain (%s): %s", d.Id(), err)
 	}
 
-	// an API issue will be raised in ForceNew scene, so wait for a while
-	time.Sleep(3 * time.Second) // lintignore:R018
+	if err := waitingForDomainDeleted(ctx, cdnClient, d, d.Timeout(schema.TimeoutDelete), opts); err != nil {
+		return diag.Errorf("error waiting for CDN domain (%s) deletion to complete: %s", d.Id(), err)
+	}
 	return nil
 }
 
+func waitingForDomainDeleted(ctx context.Context, client *golangsdk.ServiceClient, d *schema.ResourceData,
+	timeout time.Duration, opts *domains.ExtensionOpts) error {
+	domainName := d.Get("name").(string)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"PENDING"},
+		Target:  []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			domain, err := domains.GetByName(client, domainName, opts).Extract()
+			if err != nil {
+				parseErr := parseDeleteDetailResponseError(err)
+				if _, ok := parseErr.(golangsdk.ErrDefault404); ok {
+					return "success", "COMPLETED", nil
+				}
+				return nil, "ERROR", err
+			}
+
+			if domain == nil {
+				return nil, "ERROR", fmt.Errorf("error retrieving CDN domain: Domain is not found in API response")
+			}
+
+			status := domain.DomainStatus
+			if status == "deleting" {
```

Comment:
how about just returning PENDING directly if there's no 404 error, we don't need to be specific to `deleting` status.

Reply:
OK